client-web/src/exlibris-app/menu/menu-item.html

```diff
@@ -0,0 +1,107 @@
+
+            .item {
+                cursor: pointer;
+                margin: 4%;
+                align-items: center;
+                color: var(--menu-item-title-color, black);
+                border-radius: 3px;
+                background-color: var(--menu-item-background-color, white);
+                line-height: 40px;
+                --iron-icon-height: 26px;
+                --iron-icon-width: 26px;
+                padding-left: 12px;
+                font-size: 15px;
+            }
+
+            .name {
+                padding-left: 12px;
+            }
+
+            .count-badge {
+                position: fixed;
+                right: 10%;
+            }
+        </style>
+
+        <div id="menu-item-body" class="item" on-click="_handleClick">
+            <slot name="icon">
+                <iron-icon icon="book"
+                           style="--iron-icon-fill-color: var(--google-grey-700)"></iron-icon>
+            </slot>
+            <span id="menu-item-title" class="name">[[title]]</span>
+            <span id="menu-item-badge" class="count-badge">[[count]]</span>
+        </div>
+
+    </template>
+
+    <script>
+        /**
+         * The single item of menu list.
+         * Provides the clickable area with title. Can be customized with
+         * icon and the count number.
+         *
+         * The following custom properties are also available for styling:
+         *
+         * Custom property | Description | Default
+         * ----------------|-------------|----------
+         * `--menu-item-title-color` | Element text color | black
+         * `--menu-item-background-color` | Background color of the element display area | white
+         *
+         * @customElement
+         * @polymer
+         */
+        class MenuItem extends Polymer.Element {
+            static get is() {
+                return 'menu-item';
+            }
+
+            static get properties() {
+                return {
+                    /**
+                     * Fired on `element` display area mouse click event.
+                     *
+                     * @event menu-item-click
+                     * @param {detail} the link to redirect application.
+                     */
+
+                    /**
+                     * The link attached to the `element` click event.
+                     */
+                    link: String,
+
+                    /**
+                     * The element title to display.
+                     */
+                    title: String,
+
+                    /**
+                     * The element count badge to display.
+                     */
+                    count: Number
+                };
+            }
+
+            /**
+             * Handles user click on item display area.
+             *
+             * @private
+             */
+            _handleClick() {
+                this.dispatchEvent(new CustomEvent('menu-item-click', {detail: this.link}));
```

Comment:
This seems redundant. We can manage clicking outside.

Reply:
If I handle click from the outside, its event does not contain data about the link.  Addressing to the event.path does not give stable results, sometimes this element is `undefined`.